src/network/networkd-json.c

```diff
@@ -1014,6 +1016,39 @@ static int dhcp_server_append_json(Link *link, JsonVariant **v) {
         return json_append_one(v, "DHCPServer", w);
 }
 
+static int dhcp6_client_private_options_append_json(Link *link, JsonVariant **v) {
+        _cleanup_(json_variant_unrefp) JsonVariant *array = NULL;
+        _cleanup_free_ char *hex_buf = NULL;
+        int r;
+
+        assert(link);
+        assert(v);
+
+        if (!link->dhcp6_lease)
+                return 0;
+
+        LIST_FOREACH(options, option, link->dhcp6_lease->private_options) {
+                _cleanup_(json_variant_unrefp) JsonVariant *e = NULL;
+                char key[STRLEN("Option_000")+1];
+
+                xsprintf(key, "OPTION_%" PRIu8, option->tag);
+                hex_buf = hexmem(option->data, option->length);
+                if (!hex_buf)
+                        return -ENOMEM;
+
+                r = json_build(&e,
+                               JSON_BUILD_OBJECT(
+                                               JSON_BUILD_PAIR_STRING(key, hex_buf)));
```

Comment:
use the JSON_BUILD_PAIR_HEX() option which means you don't need to call hexmem() first.

Reply:
that said, I think it would make more sense to encode this in base64 actually, it's usually how we encode arbitrary binary blobs in json. i.e. JSON_BUILD_PAIR_BASE64()

Reply:
Wouldn't be fine to keep it in hex itself instead of encoding to base64 ?